pkg/provider/gcp/secretmanager/secretsmanager_workload_identity.go

```diff
@@ -257,3 +262,14 @@ func (g *gcpIDBindTokenGenerator) Generate(ctx context.Context, client *http.Cli
 	}
 	return idBindToken, nil
 }
+
+func clusterProjectID(store esv1beta1.GenericStore) (string, error) {
+	spec := store.GetSpec()
+	if spec.Provider.GCPSM.Auth.WorkloadIdentity.ClusterProjectID != "" {
+		return spec.Provider.GCPSM.Auth.WorkloadIdentity.ClusterProjectID, nil
+	} else if spec.Provider.GCPSM.ProjectID != "" {
+		return spec.Provider.GCPSM.ProjectID, nil
+	} else {
+		return "", fmt.Errorf(errNoProjectID)
+	}
+}
```

Comment:
Wouldn't it make more sense to get the spec directly as an input argument?
I believe this would be possible b/c a spec object is already being generated within `TokenSource` method.

Reply:
yeah that should be possible, i'll update

Reply:
updated